cf-custom-resources/lib/custom-domain-app-runner.js

```diff
@@ -0,0 +1,189 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/* jshint node: true */
+/*jshint esversion: 8 */
+
+"use strict";
+
+const AWS = require('aws-sdk');
+const { report, defaultSleep } = require('../lib/partials');
+
+const ERR_NAME_CUSTOM_DOMAIN_ALREADY_ASSOCIATED = "CustomDomainAlreadyAssociatedException";
+const ERR_NAME_INVALID_REQUEST = "InvalidRequestException";
+const DOMAIN_STATUS_PENDING_VERIFICATION = "pending_certificate_dns_validation";
+const ATTEMPTS = 10;
+
+
+let sleep = defaultSleep;
+let appRoute53Client, appRunnerClient, appHostedZoneID;
+
+exports.handler = async function (event, context) {
+    const props = event.ResourceProperties;
+    const [serviceARN, appDNSRole, customDomain] = [props.ServiceARN, props.AppDNSRole, props.CustomDomain,];
+    appHostedZoneID = props.HostedZoneID;
+
+    // Configure clients.
+    appRoute53Client = new AWS.Route53({
+        credentials: new AWS.ChainableTemporaryCredentials({
+            params: { RoleArn: appDNSRole, },
+            masterCredentials: new AWS.EnvironmentCredentials("AWS"),
+        }),
+    });
+    appRunnerClient = new AWS.AppRunner();
+
+    let addCustomDomainErr;
+    await addCustomDomain(serviceARN, customDomain).catch(async err => {
+        addCustomDomainErr = err;
+        if (err.name === "CustomDomainAlreadyAssociated") {
+            console.log("Custom domain already associated. Do nothing.");
+            return;
+        }
+        console.log(`Caught error: ${err.message}`);
+        await report(event, context, "FAILED", event.LogicalResourceId, null, err.message).catch((err) => {
+            throw new Error("send response: " + err.message);
+        });
+    });
+
+    if (!addCustomDomainErr) {
+        console.log("Finished");
+        await report(event, context, "SUCCESS", event.LogicalResourceId).catch((err) => {
+            throw new Error("send response: " + err.message);
+        });
```

Comment:
I don't think you have to catch these, in `report` we have: `.on("error", reject)` so it will already throw an error